tests/flow_control.rs

```diff
@@ -530,7 +536,16 @@ fn recv_window_update_on_stream_closed_by_data_frame() {
             stream.send_data("hello".into(), true).unwrap();
 
             // Wait for the connection to close
-            h2.unwrap()
+            h2.map(|h2| {
+                // keep `stream` from being dropped in order to prevent
+                // it from sending an RST_STREAM frame.
+                std::mem::forget(stream);
+                // i know this is kind of evil, but it's necessary to
+                // ensure that the stream is closed by the EOS frame,
+                // and not by the RST_STREAM.
```

Comment:
Would be better to just wait on receiving the frame back from the server instead.

Reply:
I tried `h2.drive(stream)` but that doesn't seem to be triggered by the window update & never terminates. Am I missing something?

Reply:
Hm, can we receive a response from the server then?

Reply:
@seanmonstar sorry, how would you suggest I do that, exactly? Nothing I've tried here has worked so far...

Reply:
Can the mock `srv` just send a headers frame after the window update?

Reply:
Oh, sure! I wasn't sure if it was okay for me to modify the mock server's behavior in these tests...this could've been resolved a long time ago if I'd just asked about that, sorry.

Reply:
UPDATE: that doesn't seem to work either; sending the EOS frame closes the mock, so it panics with
```

Reply:
Can't make it work in the prioritization test either, it seems to wait forever?

Reply:
Hm, OK, that's fine. I didn't love the `forget`, but getting around it seems to be annoying, and we don't yet have a way to wait on the `stream` to receive window updates, so that can't be used.